### ISSUE TITLE
`cf help` should behave like `cf help -a` when STDOUT is being piped

### ISSUE THREAD

**Issue by wendorf** [cloudfoundry/cli#1013]
* Command  
`cf help | grep bindings`

* What occurred, incl. error and/or stack trace
Nothing was printed on my terminal

* What you expected to occur
Something (in this case, the equivalent of `cf help -a | grep bindings`) was printed on my terminal. Because I am piping the output, I most likely want the full -a output, so this should be detected and enabled.

* CLI Version  
6.22.2+a95e24c-2016-10-27

* Platform & Shell Details  
MacOS 10.12.1 iTerm 2 3.0.12

**Comment by dkoper** [comment]
Hi @wendorf 

Interesting suggestion!
I've discussed this with our designers. I feel the value of assuming that the user wants all when piping (the "80%") does not outway the confusion when they don't (e.g. when doing a `cf h | more` to *reduce* the amount of output or `cf h > common-help.txt` for whatever reason.

I noticed `git --help | more` (`git` is one of the commands we used for reference when redesigning the help as it addressed the issue we had with long help text and is a command most CF users are familiar with) doesn't do this either. Have you seen this with any other common commands?

I think what many existing users of the cf CLI, including myself, are experiencing is that `cf h | grep ...` is in our fingers and it takes a few times to get used to adding the `-a`. That said, the feedback I get is that it doesn't take that long (YMMV ;)) to get used to.

Regards,
Dies Koper
CF CLI PM
